app/models/repository/access.rb

```diff
@@ -4,6 +4,24 @@ module Repository::Access
   OPTIONS = %w[public_r public_rw private_r private_rw]
   DEFAULT_OPTION = OPTIONS[0]
 
+  def self.as_read_only(access)
+    access.split('_').first + '_r'
+  end
+
+  def destroy_expired_access_tokens
+    access_tokens.select(&:expired?).map(&:destroy)
```

Comment:
An old habit from using Haskell: "Want to do something with all the elements? Use map." ;)
I will change this to `each`.